junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertionsTests.kt

```diff
@@ -211,6 +211,59 @@ class KotlinAssertionsTests {
         assertMessageStartsWith(error, assertionMessage)
     }
 
+    @Test
+    fun `assertNotNull with compiler smart cast`() {
+        val nullableString: String? = "string"
+
+        assertNotNull(nullableString)
+        assertFalse(nullableString.isEmpty()) // smart cast to a non nullable object
+    }
+
+    @Test
+    fun `assertNull with compiler smart cast`() {
+        val nullableString: String? = null
+
+        assertNull(nullableString)
+        // even safe call is not allowed, because compiler knows that string is always null
+        // nullableString?.isEmpty()
```

Comment:
```suggestion
        // Even safe call is not allowed, because compiler knows that nullableString is always null.
        // nullableString?.isEmpty()
```
just a nit, I've been converted a few years back after reading [this post](https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html).

If you agree, please review your comments (including code samples in KDoc).

Reply:
I see there is a variety of comment styles across the codebase. But let's try writing sentences :)